File: SYP.Web/Modules/ServerTypes/Warehouse/StockEntriesForm.ts
import { DateTimeEditor, EnumEditor, initFormType, LookupEditor, MultipleImageUploadEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";
import { StockEntryDetailsEditor } from "../../Warehouse/StockEntryDetails/StockEntryDetailsEditor";
import { StockEntryStatus } from "../_Ext/StockEntryStatus";

export interface StockEntriesForm {
    EntryNo: StringEditor;
    WarehouseId: LookupEditor;
    EntryDate: DateTimeEditor;
    Status: EnumEditor;
    Description: StringEditor;
    DetailList: StockEntryDetailsEditor;
    Attachments: MultipleImageUploadEditor;
}

export class StockEntriesForm extends PrefixedContext {
    static readonly formKey = 'Warehouse.StockEntries';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!StockEntriesForm.init) {
            StockEntriesForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = DateTimeEditor;
            var w3 = EnumEditor;
            var w4 = StockEntryDetailsEditor;
            var w5 = MultipleImageUploadEditor;

            initFormType(StockEntriesForm, [
                'EntryNo', w0,
                'WarehouseId', w1,
                'EntryDate', w2,
                'Status', w3,
                'Description', w0,
                'DetailList', w4,
                'Attachments', w5
            ]);
        }
    }
}

[StockEntryStatus]; // referenced types

File: SYP.Web/Modules/Warehouse/StockEntries/StockEntriesForm.cs

namespace SYP.Warehouse.Forms;

[FormScript("Warehouse.StockEntries")]
[BasedOnRow(typeof(StockEntriesRow), CheckNames = true)]
public class StockEntriesForm
{
    [Tab("General Information")]
    [HalfWidth]
    public string EntryNo { get; set; }

    [HalfWidth]
    public int WarehouseId { get; set; }

    [HalfWidth, DateTimeEditor]
    public DateTime EntryDate { get; set; }

    [HalfWidth]
    public StockEntryStatus Status { get; set; }

    [FullWidth]
    public string Description { get; set; }

    [Tab("Products")]
    [StockEntryDetailsEditor, IgnoreName]
    public List<StockEntryDetailsRow> DetailList { get; set; }

    [Tab("Files")]
    [FullWidth]
    public string Attachments { get; set; }
}
